android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/ContactsModule.java

```diff
@@ -187,408 +258,307 @@ public void getContactsAsync(final ReadableMap options, final Promise promise) {
+        contact.relationships.add(new Contact.RelationshipItem(cursor));
       }
     }
-    return phoneNumbers;
+      return map;
   }
 
-  private WritableArray getAddressesFromContentResolver(long id, ContentResolver cr) {
-    WritableArray addresses = Arguments.createArray();
-    Cursor cursor = cr.query(
-        CommonDataKinds.StructuredPostal.CONTENT_URI,
-        null,
-        CommonDataKinds.StructuredPostal.CONTACT_ID + " = ?",
-        new String[] { Long.toString(id) },
-        null
-    );
-
-    if (cursor != null) {
-      try {
-        while (cursor.moveToNext()) {
-          String street = cursor.getString(cursor.getColumnIndex(CommonDataKinds.StructuredPostal.STREET));
-          String city = cursor.getString(cursor.getColumnIndex(CommonDataKinds.StructuredPostal.CITY));
-          String country = cursor.getString(cursor.getColumnIndex(CommonDataKinds.StructuredPostal.COUNTRY));
-          String region = cursor.getString(cursor.getColumnIndex(CommonDataKinds.StructuredPostal.REGION));
-          String neighborhood = cursor.getString(cursor.getColumnIndex(CommonDataKinds.StructuredPostal.NEIGHBORHOOD));
-          String postcode = cursor.getString(cursor.getColumnIndex(CommonDataKinds.StructuredPostal.POSTCODE));
-          String poBox = cursor.getString(cursor.getColumnIndex(CommonDataKinds.StructuredPostal.POBOX));
-          int type = cursor.getInt(cursor.getColumnIndex(CommonDataKinds.StructuredPostal.TYPE));
-
-          if (street == null && city == null && country == null && region == null && neighborhood == null && postcode == null && poBox == null) {
-            return null;
-          }
-
-          WritableMap details = Arguments.createMap();
 
-          if (street != null) {
-            details.putString("street", street);
-          }
-          if (city != null) {
-            details.putString("city", city);
-          }
-          if (country != null) {
-            details.putString("country", country);
-          }
-          if (region != null) {
-            details.putString("region", region);
-          }
-          if (neighborhood != null) {
-            details.putString("neighborhood", neighborhood);
-          }
-          if (postcode != null) {
-            details.putString("postcode", postcode);
-          }
-          if (poBox != null) {
-            details.putString("poBox", poBox);
-          }
+  private boolean isMissingPermissions() {
+    return Build.VERSION.SDK_INT >= 23 &&
+        ContextCompat.checkSelfPermission(
+            getReactApplicationContext(),
+            Manifest.permission.READ_CONTACTS) != PackageManager.PERMISSION_GRANTED;
+  }
 
-          String label;
+  private static class Contact {
+    private String contactId;
+    private String displayName;
+    private String givenName = "";
+    private String middleName = "";
+    private String familyName = "";
+    private String prefix = "";
```

Comment:
unused
```
if (!TextUtils.isEmpty(prefix)) {
        contact.putString("namePrefix", prefix);
      }
```
🙂

Reply:
check.

good catch, hate not being able to seed data on tests 😠